internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -265,6 +265,13 @@ func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 		deployment.Spec.Replicas = nil
 	}
 
+	// apply merge patch to deployment
+	if merged, err := deploymentConfig.ApplyMergePatch(deployment); err == nil {
+		deployment = merged
+	} else {
+		return nil, err
```

Comment:
if there is a bad patch, we shouldnt fail the creation of deployment, imo we should not patch the deployment and populate the error in the GatewayClass status. Updating the status can be a followup

Reply:
Good point, updated.